util/ioutils/multireadcloser.go

```diff
@@ -0,0 +1,48 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package ioutils
+
+import (
+	"errors"
+	"io"
+)
+
+type MultiReaderCloser struct {
+	c []io.ReadCloser
+	io.Reader
+}
+
+func NewMultiReadCloser(rcs []io.ReadCloser) *MultiReaderCloser {
```

Comment:
Any particular reason we don't just use the exact readclosers given to us?

Reply:
Because `io.MultiReader` takes an `[]io.Reader`. `[]io.ReadCloser` is not the same type as `[]io.Reader` and can't be used in its place without this sort of conversion.